src/traces/scattermapbox/convert.js

```diff
@@ -0,0 +1,393 @@
+/**
+* Copyright 2012-2016, Plotly, Inc.
+* All rights reserved.
+*
+* This source code is licensed under the MIT license found in the
+* LICENSE file in the root directory of this source tree.
+*/
+
+
+'use strict';
+
+var Lib = require('../../lib');
+var subTypes = require('../scatter/subtypes');
+
+var COLOR_PROP = 'circle-color';
+var SIZE_PROP = 'circle-radius';
+
+
+module.exports = function convert(calcTrace) {
```

Comment:
The `convert` step here essentailly converts `gd.calcdata` to `mapbox-gl` paint and layer properties (see reference [guide](https://www.mapbox.com/mapbox-gl-style-spec/))